### PR TITLE
Validating data records in aggregation process

### PR DESCRIPTION
**Pull request by srikartati**
Before aggregating, valid data records if all the records have
proper values after decoding.

### CODE REVIEW

File: pkg/intermediate/aggregate.go [hunk 1]
```diff
@@ -110,15 +110,26 @@ func (a *AggregationProcess) AggregateMsgByFlowKey(message *entities.Message) er
...
-		flowKey, err := getFlowKeyFromRecord(record)
-		if err != nil {
-			return err
-		}
-		if err = a.addOrUpdateRecordInMap(flowKey, record); err != nil {
-			return err
+		// Validate the data record. If invalid, we log the error and move to the next
+		// record.
+		if !validateDataRecord(record) {
+			klog.V(2).Infof("Invalid data record because decoded values of elements are not valid.")
```
Comment (zyiou) [inline pkg/intermediate/aggregate.go:118]: seems it is more an `Errorf` than `Infof`

**Review by zyiou — COMMENTED** [review]
Overall LGTM. I will fix the tls certificate issue in the unit tests.

File: pkg/intermediate/aggregate.go (hunk 1 above)
Comment (srikartati) [reply pkg/intermediate/aggregate.go:118]: Yes `Errorf` is more appropriate. The intention was to use error log but missed it.

**Review by zyiou — APPROVED** [review]
LGTM

**Pull request merged**